### PR TITLE
liquidbase, soft delete, search implemented v3.0

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/resources/db/changelog/changes/02-add-test-book.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+databaseChangeLog:
+  - changeSet:
+      id: insertBooks
```
Comment (JeniaSan) [inline src/main/resources/db/changelog/changes/02-add-test-book.yaml:3]: Do you really have to create a file to add a book to database?What if you have to add 10 000 books in your DB?

**Review by JeniaSan — CHANGES_REQUESTED** [review]
Nice solution, let's improve it

File: src/main/resources/db/changelog/changes/02-add-test-book.yaml (hunk 1 above)
Comment (CodeWebWeaver) [reply src/main/resources/db/changelog/changes/02-add-test-book.yaml:3]: Then I write method that add books. It is just test that defines liquibase work correct

**Review by JeniaSan — APPROVED** [review]
LGTM!

File: src/main/java/com/parkhomovsky/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -1,24 +1,36 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 @RequiredArgsConstructor
 @RestController
-@RequestMapping(value = "/api/books")
+@RequestMapping(value = "/books")
```
Comment (vovasalyha) [inline src/main/java/com/parkhomovsky/bookstore/controller/BookController.java:22]:
```suggestion
@RequestMapping(value = "/api/books")
```

File: src/main/java/com/parkhomovsky/bookstore/repository/book/BookSpecificationProviderManager.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(b -> b.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() ->
+                        new RuntimeException("Can`t find correct specification provider for key: "
```
Comment (vovasalyha) [inline src/main/java/com/parkhomovsky/bookstore/repository/book/BookSpecificationProviderManager.java]: better throw `NoSuchElement`

File: src/main/resources/db/changelog/changes/02-add-test-book.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                  value: "14.56"
+              - column:
+                  name: description
+                  value: "Test book description"
+              - column:
+                  name: cover_image
+                  value: "URL"
+              - column:
+                  name: is_deleted
+                  valueBoolean: "false"
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/02-add-test-book.yaml]: empty line

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Left few suggestions.

File: src/main/java/com/parkhomovsky/bookstore/controller/BookController.java (hunk 2 above)
Comment (CodeWebWeaver) [reply src/main/java/com/parkhomovsky/bookstore/controller/BookController.java:22]: But I have this in application.properties:
server.servlet.context-path=/api

Comment (vovasalyha) [reply src/main/java/com/parkhomovsky/bookstore/controller/BookController.java:22]: good, didn't notice that.

File: src/main/java/com/parkhomovsky/bookstore/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
+import org.hibernate.validator.constraints.URL;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull(message = "Title cannot be null")
```
Comment (vovasalyha) [inline src/main/java/com/parkhomovsky/bookstore/dto/CreateBookRequestDto.java:13]: If I'm not mistaken, validation is not related to this task, but to the other.

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**